cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -189,8 +189,11 @@ private Channel connectViaCliPort(URL jenkins, CliPort clip) throws IOException
                 rsp.write(ch);
             }
             String head = new BufferedReader(new StringReader(rsp.toString("ISO-8859-1"))).readLine();
-            if (!head.startsWith("HTTP/1.0 200 "))
-                throw new IOException("Failed to establish a connection through HTTP proxy: "+rsp);
+            if (!(head.startsWith("HTTP/1.0 200 ") || head.startsWith("HTTP/1.1 200 "))) {
+                s.close();
+                System.err.println("Failed to tunnel the CLI port through the HTTP proxy. Falling back to HTTP.");
```

Comment:
I did this to be consistent with other part of the code, like https://github.com/jenkinsci/jenkins/blob/master/cli/src/main/java/hudson/cli/CLI.java#L418 or https://github.com/jenkinsci/jenkins/blob/master/cli/src/main/java/hudson/cli/CLI.java#L482 . I don't mind outputting to stdout if you prefer